src/ServiceControl.Audit/Auditing/AuditPersister.cs

```diff
@@ -89,12 +73,11 @@ public async Task<IReadOnlyList<MessageContext>> Persist(IReadOnlyList<MessageCo
                             Logger.Debug("Adding audit message for bulk storage");
                         }
 
-                        using (auditBulkInsertDurationMeter.Measure())
-                        {
-                            await unitOfWork.RecordProcessedMessage(processedMessage, context.Body);
-                        }
+                        var auditSw = Stopwatch.StartNew();
+                        await unitOfWork.RecordProcessedMessage(processedMessage, context.Body);
+                        auditBulkInsertDuration.Record(auditSw.ElapsedMilliseconds);
```

Comment:
There are a lot of things done in this method. Even multiple measurements.
I think we should consider doing the measures in the `unitOfWork` methods itself. The measurement is then simply that whole method and isn't making this method even worse then it already is :-)